Library/Homebrew/unpack_strategy.rb

```diff
@@ -67,6 +67,30 @@ def extract(to: nil, basename: nil)
     unpack_dir.mkpath
     extract_to_dir(unpack_dir, basename: basename)
   end
+
+  def extract_nestedly(to: nil, basename: nil)
+    if self.is_a?(UncompressedUnpackStrategy)
+      extract(to: to, basename: basename)
+      return
+    end
+
+    Dir.mktmpdir do |tmp_unpack_dir|
+      tmp_unpack_dir = Pathname(tmp_unpack_dir)
+
+      extract(to: tmp_unpack_dir, basename: basename)
+
+      children = tmp_unpack_dir.children
+
+      if children.count == 1
+        s = self.class.detect(children.first)
+
+        s.extract_nestedly(to: to, basename: basename)
```

Comment:
Any potential for infinite recursion here?

Reply:
I don't think so, since the recursion will stop when `self` is a `UncompressedUnpackStrategy`, which is the default.

Although I think we should not recurse into nested directories.